src/products/services/products/products.service.spec.ts

```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
         .mockReturnValue({
           exec: jest.fn().mockResolvedValueOnce(deletedProduct)
         } as any);
-      const result = await service.delete(mockProduct.id);
-      expect(model.findByIdAndUpdate).toHaveBeenCalledWith(
-        mockProduct.id,
-        { isHidden: true }
-      );
-      expect(result).toEqual(deletedProduct);
+      mockProduct = await service.delete(mockProduct.id);
```

Comment:
You want to leave this as before, `const result = await service.delete(mockProduct.id)` as the mockProduct is already defined and we want to see what the service brings back.